legend-engine-xt-serviceStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/service/ServiceExecutor.java

```diff
@@ -59,11 +61,17 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 
 public class ServiceExecutor
 {
-    public static InputStreamResult executeHttpService(String url, List<Header> headers, StringEntity requestBodyEntity, HttpMethod httpMethod, String mimeType, List<SecurityScheme> securitySchemes, MutableList<CommonProfile> profiles)
+    private final ServiceStoreExecutionConfiguration serviceStoreExecutionConfiguration;
+
+    public ServiceExecutor(ServiceStoreExecutionConfiguration serviceStoreExecutionConfiguration)
```

Comment:
We should not pass config directly rather pass only stuff we need - like credential provider